src/AltinnCore/Common/Services/Implementation/DataAppSI.cs

```diff
@@ -75,55 +97,72 @@ public async Task<Instance> InsertData<T>(T dataToSerialize, Guid instanceId, Ty
         /// <inheritdoc />
         public void UpdateData<T>(T dataToSerialize, Guid instanceId, Type type, string applicationOwnerId, string applicationId, int instanceOwnerId, Guid dataId)
         {
-            string apiUrl = $"{_platformSettings.GetApiStorageEndpoint}instances/{instanceId}/data/{dataId}?instanceOwnerId={instanceOwnerId}";
+            string apiUrl = $"instances/{instanceId}/data/{dataId}?instanceOwnerId={instanceOwnerId}";
+            string token = JwtTokenUtil.GetTokenFromContext(_httpContextAccessor.HttpContext, _cookieOptions.Cookie.Name);
+            _client.DefaultRequestHeaders.Accept.Clear();
 
-            using (HttpClient client = new HttpClient())
+            if (_client.DefaultRequestHeaders.Contains("Authorization"))
```

Comment:
if this piece of code is common to all the clients, can we move it to the accessor?

Reply:
But what if the token has changed between two times the client is used? Will the token be the same as for the initial session the whole time, or isn't this a real issue ?

Reply:
setting the header for each request, but extracted the functionality